oci/layout/oci_transport.go

```diff
@@ -189,14 +191,28 @@ func (ref ociReference) NewImage(ctx *types.SystemContext) (types.Image, error)
 	return image.FromSource(src)
 }
 
-func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
+func (ref ociReference) getIndex() (imgspecv1.Index, error) {
+	index := imgspecv1.Index{
+		Versioned: imgspec.Versioned{
+			SchemaVersion: 2,
+		},
+	}
+
 	indexJSON, err := os.Open(ref.indexPath())
+	defer indexJSON.Close()
```

Comment:
Moving the `defer` up is incorrect; `indexJSON` may be `nil` at this point.

Reply:
true